### PR TITLE
fix: PR indexes & api setup and teardown

### PR DESCRIPTION
**Pull request by Riccardo-Brown**
PR adds back setting up api and index based on `crn-create-environment` or `gp2-create-environment` label. That was added in https://github.com/yldio/asap-hub/pull/3862 and reverted

Also fixes the following issue

### Problem
PR that did not have dedicated environment were deleting `asap-hub_dev` and `gp2-hub_dev` indexes during PR clean up workflow

### Solution
only run the delete index step if the index is not dev or prod
only remove webhook if it is PR environment
only copy index if it is a PR environment

### CODE REVIEW

File: .github/workflows/on-remove.yml
Comment (peterstarling) [inline .github/workflows/on-remove.yml:166]: isn't this a little too specific, shouldn't we rely on the environment or label or something instead? like if we ever change the naming convention it would be good if this still worked as expected.

Plus - if you really are going to be excluding the dev index you might as well add prod in there too just to be safe

**Review by peterstarling — CHANGES_REQUESTED** [review]

**Review by peterstarling — CHANGES_REQUESTED** [review]
i think you should have a look at how we currently set up contentful environment in on-push-branch workflow - you will see that we already have a way to decide if a dedicated env is used there and i think it should be expanded for algolia. Your changes would suggest that contentful never work on branches but it did

**Comment by peterstarling** [comment]
Check out how its currently done for Contentful:

https://github.com/yldio/asap-hub/blob/708ab7a97ba2267bc3e24ec19a455e8ffca1d627/.github/workflows/on-push-branch.yml#L31-L40

...is how we get the current state of the environment

https://github.com/yldio/asap-hub/blob/708ab7a97ba2267bc3e24ec19a455e8ffca1d627/.github/workflows/on-push-branch.yml#L96

... and that's how it is then passed on to create-environment workflow. You should follow this approach for algolia.

File: .github/workflows/on-remove.yml
Comment (gabiayako) [inline .github/workflows/on-remove.yml:1]: Shouldn't we add the condition to run if it is a PR env on `Contentful Environment Teardown` in both `crn-contentful-remove` and `gp2-contentful-remove` so we don't delete `Development` environment?

File: .github/workflows/reusable-create-environment.yml
Comment (gabiayako) [inline .github/workflows/reusable-create-environment.yml:221]:
```suggestion
        if: ${{ inputs.environment-name=='Branch' && steps.index-exists.outputs.index-exists == 'false' && inputs.gp2-on-branch-env == 'true' }}
```

File: .github/workflows/on-remove.yml
Comment (Riccardo-Brown) [reply .github/workflows/on-remove.yml:1]: the condition is in the script https://github.com/yldio/asap-hub/blob/master/packages/contentful/scripts/teardown.sh

Comment (gabiayako) [reply .github/workflows/on-remove.yml:1]: hmm ok, I thought it was deleted before when the PR got merged the first time, but I think I was wrong then :)

**Review by gabiayako — APPROVED** [review]

**Review by markcsims — APPROVED** [review]

**Review by peterstarling — APPROVED** [review]

**Pull request merged**
